src/pages/Trade/components/ChartCard/Chart.tsx

```diff
@@ -1,50 +1,180 @@
-import React, { FC, useContext } from 'react';
-import { connect } from 'react-redux';
+import React, { FC, useContext, useEffect, useMemo } from 'react';
+import { ConnectedProps, connect } from 'react-redux';
 import styled, { ThemeContext } from 'styled-components';
 import { AreaChart, XAxis, YAxis, Area, Tooltip } from 'recharts';
 import format from 'date-fns/format';
 import { useTranslation } from 'react-i18next';
 import isNumber from 'lodash/isNumber';
+import find from 'lodash/find';
 
 import RechartsResponsiveContainer from 'components/RechartsResponsiveContainer';
 
 import { DataLarge } from 'components/Typography';
 import Spinner from 'components/Spinner';
 
-import { formatCurrencyWithPrecision } from 'utils/formatters';
+import useInterval from 'shared/hooks/useInterval';
+import { formatCurrencyWithSign } from 'utils/formatters';
 
 import { RootState } from 'ducks/types';
-import { SynthPair, SynthDefinitionMap, getAvailableSynthsMap } from 'ducks/synths';
+import { getIsWalletConnected } from 'ducks/wallet/walletDetails';
+import { fetchMyTradesRequest, getMyTrades } from 'ducks/trades/myTrades';
+import { HistoricalTrade } from 'ducks/trades/types';
+import { SynthPair, getAvailableSynthsMap } from 'ducks/synths';
 import { PeriodLabel, PERIOD_IN_HOURS } from 'constants/period';
+import { DEFAULT_REQUEST_REFRESH_INTERVAL } from 'constants/ui';
 
 import { ChartData } from './types';
 
-type StateProps = {
-	synthsMap: SynthDefinitionMap;
+const mapStateToProps = (state: RootState) => ({
+	synthsMap: getAvailableSynthsMap(state),
+	trades: getMyTrades(state),
+	isWalletConnected: getIsWalletConnected(state),
+});
+
+const mapDispatchToProps = {
+	fetchMyTradesRequest,
 };
 
-type Props = {
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
+
+type ChartProps = PropsFromRedux & {
 	data: ChartData;
 	isLoading: boolean;
 	period: PeriodLabel;
 	synthPair: SynthPair;
 };
-type ChartProps = StateProps & Props;
 
-const Chart: FC<ChartProps> = ({ synthPair: { quote }, data, isLoading, period, synthsMap }) => {
+const Chart: FC<ChartProps> = ({
+	isWalletConnected,
+	fetchMyTradesRequest,
+	trades,
+	synthPair: { quote },
+	data,
+	isLoading,
+	period,
+	synthsMap,
+}) => {
 	const { colors } = useContext(ThemeContext);
 	const { t } = useTranslation();
 	const synthSign = synthsMap[quote.name] && synthsMap[quote.name].sign;
 
+	useEffect(() => {
+		if (isWalletConnected) {
+			fetchMyTradesRequest();
+		}
+	}, [isWalletConnected, fetchMyTradesRequest]);
+
+	const intervalOrNull = isWalletConnected ? DEFAULT_REQUEST_REFRESH_INTERVAL : null;
+
+	useInterval(() => {
+		fetchMyTradesRequest();
```

Comment:
@dvd-schwrtz @evgenyboxer do we need an interval? We're already tracking the exchange event in GlobalEventsGate.tsx.

```
		Synthetix.contract.on(EXCHANGE_EVENTS.SYNTH_EXCHANGE, (address: string) => {
			if (address === currentWallet) {
				fetchWalletBalancesRequest();
			}
		});

Reply:
I guess it's because we fetch the trades from the graph, right?

Reply:
Yes, we could theoretically run a subscription instead, but this just polls for new trades every 30 seconds while the user is logged in. It also doesn't update the chart data or trade markers if the user has not made a trade

Reply:
Instead of polling, we could add an event listener like this to refetch the trades when the user has made an exchange. Although it would probably constantly be checking every other users trades, so not sure if it would be more performant.

Reply:
@clementbalestrat yeah exactly, while the contract is great for the new data coming in, we do need the historical one too.